manageperson.rb

```diff
@@ -0,0 +1,35 @@
+def create_teacher
+  print "Teacher's Specialization: "
+  specialization = gets.chomp
+  print "Teacher's age: "
+  age = gets.chomp.to_i
+  print "Teacher's name: "
+  name = gets.chomp
+
+  teacher = Teacher.new(age, name, specialization, parent_permission: true)
+  @people << teacher
+  puts 'You have successfully registered a Teacher'
+end
+
+def create_student
+  print 'Age: '
+  age = gets.chomp.to_i
+  print 'Name: '
+  name = gets.chomp
+  print 'Has parent permission? [Y/N]: '
+
+  parent_permission = gets.chomp.downcase
+
+  case parent_permission
+  when 'n'
+    student = Student.new(nil, age, name, parent_permission: false)
+    @people << student
+  when 'y'
+    student = Student.new(nil, age, name, parent_permission: true)
+    @people << student
+  else
+    'You have entered an invalid option'
+  end
```

Comment:
- [ ] @emhamza and @Jeddaa  Great work creating a module to handle the student  addition, however, I would suggest you move these line to your `create_person` method, this will ensure the app follow the `Single responsibility rule` where each function is dedicated to a specific task alone

Reply:
Please explain what you are trying to say. You current feedback is  simply confusing us! :)

Reply:
- What he means is move these two methods to the `App.rb` file since it imports all the modules in one place. This keeps your logic clean and a class is only responsible for one thing.

Reply:
@whiz25 hello, please do you mean this is the only issue we have?